File: src/app/models/trainer.ts
export class Trainer {

    constructor(    
        private _id: number,
        private _name: string,
    ) {}

    get name() { return this._name; }
    get id() { return this._id; }
    
    set name(name: string) { this._name = name; }
    set id(id: number) { this._id = id; }


    }

    export interface TrainerInterface {
        id: number;
        name: string;
    }

File: backend/src/main/java/pokemon/models/Trainer.java
package pokemon.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Trainer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @Column(unique = true)
    private String name;
    @OneToMany(mappedBy = "trainer")
    @JsonIgnore
    private List<Pokemon> pokemons;

}
